pkg/controller/volume/scheduling/scheduler_binder.go

```diff
@@ -455,6 +482,35 @@ var (
 	versioner = etcd3.APIObjectVersioner{}
 )
 
+// checkInlineVolumes ensures that the CSI driver required by each volume
+// is really running on the node.
+func (b *volumeBinder) checkInlineVolumes(nodeName string, inlineVolumes []*v1.CSIVolumeSource) (bool, error) {
```

Comment:
This is going to break integration with cluster autoscaler. Cluster autoscaler works by simulating Node objects, and there is currently no way to tell it what nodes a CSI driver is expected to be installed on. The current expectation is that a CSI driver will be installed on all nodes. I think the current way we deploy csi hostpath only on a single node is more of an anti-pattern.

Reply:
> This is going to break integration with cluster autoscaler. Cluster autoscaler works by simulating Node objects, and there is currently no way to tell it what nodes a CSI driver is expected to be installed on.

So the autoscaler simulates nodes, a pod gets scheduled onto it, and then the autoscaler actually creates the node? Yes, that won't work for pods using ephemeral inline volumes with this check active.

> The current expectation is that a CSI driver will be installed on all nodes.

Sounds like something that we should document.